### ISSUE TITLE
[a11y]: Multi Select options not readable by screen reader

### ISSUE THREAD

**Issue by dchyun** [carbon-design-system/carbon#22409]
### Package

@carbon/web-components

### Package version

2.56.0

### Automated testing tool and ruleset

manual

### Assistive technology

VoiceOver

### Description

In the Muti Select web component, the options available in the menu are not announced by a screen reader. When focusing on the component it is announced to the screen reader, but upon activating the menu and navigating through the menu items none of the menu items are announced as the user uses the arrow keys to navigate through them.

This is because the menu is not properly attached to the component's label. The menu has `aria-labelledby="dropdown-label"` but the label element has no `id` set. 

<img width="401" height="281" alt="Image" src="https://github.com/user-attachments/assets/c34ca524-e419-429e-844e-aa4e389a3c76" />

This issue is not seen in the Dropdown because the label and menu are properly connected.

<img width="354" height="323" alt="Image" src="https://github.com/user-attachments/assets/441b1956-671d-40de-8087-ad97aa452ba3" />

### WCAG 2.1 Violation

Success Criterion 4.1.2 Name, Role, Value

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-multi-select--default

### Steps to reproduce

1. Open the Multi Select components in the web component's Storybook
2. Turn on VoiceOver + macOS or other screen reader
3. Focus on the multi select
4. Open the multi-select
5. Navigate through the options available using the keyboard and ensure they are announced by the screenreader as your focus changes 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by maradwan26** [comment]
Can verify that the same issue also exists with [Dropdown](https://web-components.carbondesignsystem.com/?path=/story/components-dropdown--default), and [ComboBox](https://web-components.carbondesignsystem.com/?path=/story/components-combo-box--default)
